### DISCUSSION TITLE
Message history

### DISCUSSION THREAD

**Discussion by MichaelJoJackson** [aiko-chan-ai/DiscordBotClient#45] [Ideas]
I think it would be better if every new message sent to the bot was marked in the "Direct" list, otherwise it is always empty. In order to see the messages, you need to click on the user and go to his direct, although it would be much more convenient if everything came at once and was marked as "1 unread message".

**Comment by aiko-chan-ai** [comment]
can you explain more clearly 💀

**Reply by MichaelJoJackson** [reply]
I made a video showing what I'm talking about. I would very much like to receive messages immediately in direct, the chat should, in theory, be automatically created, but this is not so. Is it possible to add such an option? I think this is most useful for the ticket system. It would be convenient and practical, so as not to fool around with setting up and transferring messages to separate server channels. Like: I went in and immediately saw all the important messages from people that were sent to the bot.


https://drive.google.com/file/d/1bNPtLkJM1uDN3Ij0FImsFKNrhmv8Bph4/view?usp=drive_link

**Reply by aiko-chan-ai** [reply]
when logged in, bot can't see any private channels. (https://discord.com/developers/docs/change-log#june-19-2018)
after the bot is logged in, you send a message and the bot doesn't see the channel (it was a bug I fixed recently)
Please update the app 😢

**Reply by MichaelJoJackson** [reply]
That is, if the client was closed, there will be no messages after entering?

**Reply by aiko-chan-ai** [reply]
discord doesn't save bot's private channels after exiting the app

**Reply by MichaelJoJackson** [reply]
You can write a code that, after entering the client, in all chats will write 1 any message and instantly delete it. Then all chats will be in the lists and it will be possible to bypass the discord system.

**Reply by MichaelJoJackson** [reply]
I'm sorry if I'm being stupid, but maybe this will give you some idea.

**Comment by aiko-chan-ai** [comment]
> You can write a code that, after entering the client, in all chats will write 1 any message and instantly delete it. Then all chats will be in the lists and it will be possible to bypass the discord system.

spam api + rate limit

**Reply by MichaelJoJackson** [reply]
Oh, okay. Thanks a lot for the update and some clarifications, I love your project and would love to see it developed further. Please think about what I wrote above. This would immediately solve a lot of problems and troubles with unnecessary codes for the ticket system, sometimes bugs and conflicts between other functions arise due to the code. It would be like a magic wand.
